sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -1047,6 +1047,17 @@ class kernel_bundle_impl
 
   DeviceGlobalMap &getDeviceGlobalMap() { return MDeviceGlobals; }
 
+  unsigned GetKernelArgsSize(const std::string &KernelName) const {
+    auto &PM = sycl::detail::ProgramManager::getInstance();
+    const void *GlobalPointer = PM.getKernelGLobalInfoDesc(KernelName.c_str());
+    if (!GlobalPointer)
+      return 0;
+    const unsigned *SizePtr = reinterpret_cast<const unsigned *>(GlobalPointer);
+    if (!SizePtr)
+      return 0;
+    return *SizePtr;
```

Comment:
I don't think there is a case where `SizePtr` would be `nullptr` when `GlobalPointer` isn't.

```suggestion
    return *reinterpret_cast<const unsigned *>(GlobalPointer);
```